### PR TITLE
#14985: Update div_bw op

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
### Ticket
#14985, #15242

### Problem description

- Update divide backward example, documentation
- Update golden function to avoid this error
Screenshot 2024-11-18 at 9 12 39 PM

### What's changed
The following updates were made : 

- Updated bindings with `round_mode` values with acceptable values
- Updated the example for `div_bw`
- Updated golden function for `round_mode = None` to match PyTorch

### Checklist
- [x] [Post commit CI](https://github.com/tenstorrent/tt-metal/actions/runs/11949465827) - passes

Screenshot 2024-11-20 at 8 09 28 PM

### CODE REVIEW

**Review by mcw-anasuya — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/eltwise/backward/test_backward_div.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 @pytest.mark.parametrize(
     "round_mode",
     (
-        None,
+        "None",
```
Comment (eyonland) [inline tests/ttnn/unit_tests/operations/eltwise/backward/test_backward_div.py:31]: why was this necessary?

Comment (VirdhatchaniKN) [reply tests/ttnn/unit_tests/operations/eltwise/backward/test_backward_div.py:31]: In PyTorch, the div operation accepts the options None, "trunc", "floor" as valid parameters. However, "tanh" is enclosed in double quotes, while None is not. Since this is a string parameter, it would be more consistent and uniform if all options, including None, were represented within double quotes. 

In Pytorch , other string parameter ops like gelu has its inputs "none", "tanh" in quotes. Thought this change would maintain a standardised format across the API.

Comment (VirdhatchaniKN) [reply tests/ttnn/unit_tests/operations/eltwise/backward/test_backward_div.py:31]: will revert this change back and update the PR

File: tests/ttnn/unit_tests/operations/eltwise/backward/test_backward_div.py [hunk 2]
```diff
@@ -71,6 +71,23 @@ def test_bw_div_binary_default(input_shapes, device):
...
+    grad_tensor = torch.tensor([[1, 2], [3, 4]], dtype=torch.bfloat16)
+    x1_torch = torch.tensor([[1, 2], [3, 4]], dtype=torch.bfloat16, requires_grad=True)
+    x2_torch = torch.tensor([[1, 2], [3, 4]], dtype=torch.bfloat16, requires_grad=True)
+    golden_function = ttnn.get_golden_function(ttnn.div_bw)
+    golden_tensor = golden_function(grad_tensor, x1_torch, x2_torch, None)
+    grad_tt = ttnn.from_torch(grad_tensor, layout=ttnn.TILE_LAYOUT, device=device)
+    x1_tt = ttnn.from_torch(x1_torch, layout=ttnn.TILE_LAYOUT, device=device)
+    x2_tt = ttnn.from_torch(x2_torch, layout=ttnn.TILE_LAYOUT, device=device)
+    y_tt = ttnn.div_bw(grad_tt, x1_tt, x2_tt, round_mode="None")
+    tt_out_1 = ttnn.to_torch(y_tt[1])
```
Comment (eyonland) [inline tests/ttnn/unit_tests/operations/eltwise/backward/test_backward_div.py:84]: Please make this round_mode=None   (there should be no double quotes.)

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp [hunk 3]
```diff
@@ -1057,7 +1062,7 @@ void py_module(py::module& module) {
     detail::bind_binary_bw_div(
         module,
         ttnn::div_bw,
-        R"doc(Performs backward operations for divide on :attr:`input_tensor`, :attr:`alpha` or :attr:`input_tensor_a`, :attr:`input_tensor_b`, :attr:`round_mode`,  with given :attr:`grad_tensor`.)doc",
+        R"doc(Performs backward operations for divide on :attr:`input_tensor`, :attr:`alpha` or :attr:`input_tensor_a`, :attr:`input_tensor_b`, :attr:`round_mode`,  with given :attr:`grad_tensor`. :attr:`round_mode` can be  "None", "trunc", "floor".)doc",
         R"doc(BFLOAT16, BFLOAT8_B)doc");
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp:1066]: can be None, "trunc", "floor"

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by mcw-anasuya — APPROVED** [review]

**Review by KalaivaniMCW — APPROVED** [review]

File: tt-train/sources/ttml/ops/binary_ops.cpp [hunk 4]
```diff
@@ -75,7 +75,7 @@ autograd::TensorPtr operator/(const autograd::TensorPtr& a, const autograd::Tens
 
     out->set_value(ttnn::divide(a->get_value(), b->get_value()));
     autograd::GradFunction grad = [a, b, out]() {
-        auto res = ttnn::div_bw(out->get_grad(), a->get_value(), b->get_value(), "None");
+        auto res = ttnn::div_bw(out->get_grad(), a->get_value(), b->get_value(), std::nullopt);
```
Comment (eyonland) [inline tt-train/sources/ttml/ops/binary_ops.cpp:78]: Do we not default std::nullopt ?

Comment (VirdhatchaniKN) [reply tt-train/sources/ttml/ops/binary_ops.cpp:78]: yea, that's teh default value. I can remove that

**Review by eyonland — APPROVED** [review]

**Review by dmakoviichuk-tt — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/11939735849 - passed

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/11949465827 - passed

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/11968420880 - passed
